File: packages/generators/src/generators/npm/package.json.ts
import { getDnaPath, getUiPackageName, mergeStrings } from '../utils';
import { HappDefinition } from '../../types';

export default ( happ: HappDefinition) => 
`{
  "name": "${happ.name}-dev",
  "private": true,
  "workspaces": [
    "ui",
    "tests"
  ],
  "scripts": {
    "start": "npm run build:happ && cross-env HC_PORT=$(port) concurrently -k \\"npm run start:happ\\" \\"npm run start -w ${getUiPackageName(happ)}\\"",
    "test": "npm run build:happ && npm t -w tests",
    "start:happ": "hc sandbox clean && RUST_LOG=warn hc s generate ./workdir/${happ.name}.happ --run=$HC_PORT -a ${
    happ.name
  } network mdns",
    "package": "npm run build:happ && npm run package:ui && hc web-app pack workdir",
    "package:ui": "npm run build -w ${getUiPackageName(happ)} && cd ui/dist && bestzip ../dist.zip *",
    "build:happ": "npm run build:dnas && hc app pack ./workdir",
    "build:dnas": "npm run build:zomes${mergeStrings(happ.dnas.map(dna => ` && hc dna pack ./${getDnaPath(happ, dna.name)}workdir`))}",
    "build:zomes": "CARGO_TARGET_DIR=target cargo build --release --target wasm32-unknown-unknown"
  },
  "devDependencies": {
    "concurrently": "^6.2.1",
    "cross-env": "^7.0.3",
    "new-port-cli": "^1.0.0",
    "rimraf": "^3.0.2",
    "bestzip": "^2.2.0"
  }
}
`

File: crates/hc_scaffold/src/scaffold/web_app/package_json.rs
pub fn workspace_package_json(
    app_name: &String,
    ui_package_name: &String,
    web_app_workdir_path: &String,
    app_workdir_path: &String,
) -> String {
    format!(
        r#"{{
  "name": "{}-dev",
  "private": true,
  "workspaces": [
    "ui",
    "tests"
  ],
  "scripts": {{
    "start": "npm run network 2",
    "network": "hc s clean && npm run build:happ && concurrently-repeat \"npm run start:agent\"",
    "start:agent": "cross-env HC_PORT=$(port) concurrently -k \"npm run start:happ\" \"sleep 5 && npm run start -w {}\"",
    "test": "npm run build:zomes && hc app pack {} --recursive && npm t -w tests",
    "start:happ": "concurrently \"RUST_LOG=warn echo \"pass\" | hc s --piped generate {}/{}.happ --run=$HC_PORT -a {} network mdns\" \"npm run playground\"",
    "package": "npm run build:happ && npm run package -w {} && hc web-app pack {} --recursive",
    "build:happ": "npm run build:zomes && hc app pack {} --recursive",
    "build:zomes": "CARGO_TARGET_DIR=target cargo build --release --target wasm32-unknown-unknown && npm run generate:types",
    "generate:types":"rimraf ui/types && foreach -g \"./target/wasm32-unknown-unknown/release/*.wasm\" -i \"./target/wasm32-unknown-unknown/release/*_integrity.wasm\" -x \"wasm-bindgen #{{path}} --out-dir ui/types\" && rimraf ui/types/*.wasm ui/types/*_bg.wasm ui/types/*_bg.wasm.d.ts ui/types/*.js",
    "playground": "run-singleton \"holochain-playground\""
  }},
  "devDependencies": {{
    "@holochain-playground/cli": "^0.0.11",
    "concurrently": "^6.2.1",
    "concurrently-repeat": "^0.0.1",
    "cross-env": "^7.0.3",
    "foreach-cli": "^1.8.1",
    "new-port-cli": "^1.0.0",
    "rimraf": "^3.0.2",
    "run-singleton-cli": "^0.0.5"
  }},
  "engines": {{
    "npm": ">=7.0.0"
  }}
}}"#,
        app_name,
        ui_package_name,
        app_workdir_path,
        app_workdir_path,
        app_name,
        app_name,
        ui_package_name,
        web_app_workdir_path,
        app_workdir_path,
    )
}
